packages/api/src/middleware/tracking.ts

```diff
@@ -60,13 +61,40 @@ class Tracker {
     }
   }
 
-  async flushAll() {
-    const all = this.pendingUpdates;
-    this.pendingUpdates = new Map();
```

Comment:
@victorges it seemed wrong that we were re-initialising the map here because then we try to read from it in `flushLastSeen` in the loop right?